src/aatoolbox/datasources/codab/codab.py

```diff
@@ -49,50 +62,110 @@ def download(
-        >>> codab = CodAB("npl")
-        >>> npl_cod_shapefile = codab.download(
-        ...    hdx_address="administrative-bounadries-of-nepal",
-        ...    hdx_dataset_name="npl_admbnda_nd_20201117_SHP.zip")
+        >>> country_config = create_country_config(iso3="npl")
+        >>> codab = CodAB(country_config=country_config)
+        >>> npl_cod_shapefile = codab.download()
         """
-        return self._download(
+        return _download(
             filepath=self._raw_filepath,
-            hdx_address=hdx_address,
-            hdx_dataset_name=hdx_dataset_name,
+            hdx_address=self._country_config.codab.hdx_address,
+            hdx_dataset_name=self._country_config.codab.hdx_dataset_name,
             clobber=clobber,
         )
 
-    @staticmethod
-    @check_file_existence
-    def _download(
-        filepath: Path, hdx_address: str, hdx_dataset_name: str, clobber: bool
-    ):
-        return load_dataset_from_hdx(
-            hdx_address=hdx_address,
-            hdx_dataset_name=hdx_dataset_name,
-            output_filepath=filepath,
+    def load(self, admin_level: int) -> gpd.GeoDataFrame:
+        """
+        Get the COD AB data by admin level.
+
+        Parameters
+        ----------
+        admin_level: int
+            The administrative level
+
+        Returns
+        -------
+        COD AB geodataframe with specified admin level
+
+        Examples
+        --------
+        >>> from aatoolbox import create_country_config, CodAB
+        >>>
+        >>> # Retrieve admin 2 boundaries for Nepal
+        >>> country_config = create_country_config(iso3="npl")
+        >>> codab = CodAB(country_config=country_config)
+        >>> npl_admin2 = codab.load(admin_level=2)
+        """
+        admin_level_max = self._country_config.codab.admin_level_max
+        if admin_level > admin_level_max:
+            raise AttributeError(
+                f"Admin level {admin_level} requested, but maximum set to "
+                f"{admin_level_max} in {self._country_config.iso3.upper()} "
+                f"config file"
+            )
+        return self._load_admin_layer(
+            layer_name=self._country_config.codab.layer_base_name.format(
+                admin_level=admin_level
+            )
         )
 
-    def load_admin_layer(self, layer_name: str) -> gpd.GeoDataFrame:
+    def load_custom(self, custom_layer_number: int = 0) -> gpd.GeoDataFrame:
         """
-        Get an admin level by layer name.
+        Get the COD AB data from a custom (non-level) layer.
 
         Parameters
         ----------
-        layer_name: str
-            The admin layer name
+        custom_layer_number: int
+            The 0-indexed number of the layer listed in the custom_layer_names
+            parameter of the country's config file
 
         Returns
         -------
-        geopandas dataframe with COD AB admin information
+        COD AB geodataframe with custom admin level
```

Comment:
Same as above regarding the `Raises` section:

https://numpydoc.readthedocs.io/en/latest/format.html#raises